draft-ietf-masque-h3-datagram.md

```diff
@@ -303,270 +199,55 @@ Capsule Value:
 : The payload of this capsule. Its semantics are determined by the value of the
 Capsule Type field.
 
-
-## Requirements
-
-If the definition of an HTTP Method or HTTP Upgrade Token states that it uses
-the capsule protocol, its implementations MUST follow the following
-requirements:
-
-* A server MUST NOT send any Transfer-Encoding or Content-Length header fields
-  in a 2xx (Successful) response. If a client receives a Content-Length or
-  Transfer-Encoding header fields in a successful response, it MUST treat that
-  response as malformed.
-
-* A request message does not have content.
-
-* A successful response message does not have content.
-
-* Responses are not cacheable.
-
-## Intermediary Processing
-
-Intermediaries MUST operate in one of the two following modes:
-
-Pass-through mode:
-
-: In this mode, the intermediary forwards the data stream between two
-associated streams without any modification of the data stream.
-
-Participant mode:
-
-: In this mode, the intermediary terminates the data stream and parses all
-Capsule Type and Capsule Length fields it receives.
-
-Each Capsule Type determines whether it is opaque or transparent to
-intermediaries in participant mode: opaque capsules are forwarded unmodified
-while transparent ones can be parsed, added, or removed by intermediaries.
-Intermediaries MAY modify the contents of the Capsule Data field of transparent
-capsule types.
-
-Unless otherwise specified, all Capsule Types are defined as opaque to
-intermediaries. Intermediaries MUST forward all received opaque CAPSULE frames
-in their unmodified entirety. Intermediaries MUST NOT send any opaque CAPSULE
-frames other than the ones it is forwarding. All Capsule Types defined in this
-document are opaque, with the exception of the datagram capsules, see
-{{datagram-capsule}}. Definitions of new Capsule Types MAY specify that the
-newly introduced type is transparent. Intermediaries MUST treat unknown Capsule
-Types as opaque.
-
-Intermediaries respect the order of opaque CAPSULE frames: if an intermediary
-receives two opaque CAPSULE frames in a given order, it MUST forward them in
-the same order.
+Capsules MUST be forwarded unmodified by intermediaries, with the exception of
+the DATAGRAM capsule. An intermediary that understands the request semantics
+enough to know that capsules are in use MAY parse, add, or remove DATAGRAM
+capsules; see {{datagram-capsule}}. Definitions of new Capsule Types MAY specify
+optional custom intermediary processing.
```

Comment:
This paragraph still concerns me. I'm happy to acknowledge that intermediaries exist, but I dont think we should be specifying their behavior. What happens if we just delete it?

Reply:
If we don't specify intermediary handling, intermediary implementors will implement what they feel like, such as dropping unknown capsules. If we do that we can't use capsules for end to end messages like we do in WebTransport and CONNECT-IP.

Reply:
> If we don't specify intermediary handling, intermediary implementors will implement what they feel like, such as dropping unknown capsules. 

And the solution is for service operators not to use such intermediaries.


> If we do that we can't use capsules for end to end messages like we do in WebTransport and CONNECT-IP.

No, you can't do them in situations where you don't control the intermediaries.

This whole discussion of the position vis-a-vis intermediaries was a key point of disagreement on the call, so as long as that's unresolved, this PR isn't done.

Reply:
I'm not suggesting that this PR is done, resolving questions like this is why we're having this conversation :)

I see two options here:
1. the current text
2. remove mentions of intermediaries and require service operators terminate every protocol at the intermediary

I think (2) is more complex to deploy, and the cost of (1) is a few lines of spec that implementors of non-intermediaries can ignore, so I'm personally leaning towards (1).

Reply:
In today's design team meeting, we reached rough consensus to keep the current approach to intermediaries in this PR. We still have [an issue](https://github.com/ietf-wg-masque/draft-ietf-masque-connect-udp/issues/69) to track some changes that we still need to make, so I'm marking this comment thread as resolved.